CONTRIBUTING.md

```diff
@@ -55,7 +55,7 @@ You now have everything you need to start making changes!
 ### C. Merging your changes with PyBaMM
 
 10. [Test your code!](#testing)
-11. PyBaMM has online documentation at http://docs.pybamm.org/. To make sure any new methods or classes you added show up there, please read the [documentation](#documentation) section.
+11. PyBaMM has online documentation at <http://docs.pybamm.org/>. To make sure any new methods or classes you added show up there, please read the [documentation](#documentation) section.
```

Comment:
```suggestion
11. PyBaMM has online documentation at https://docs.pybamm.org/. To make sure any new methods or classes you added show up there, please read the [documentation](#documentation) section.
```

Not sure why this is needed, since this is a Markdown file? Also, we can switch to HTTPS.

Reply:
Markdownlint says that we shouldn't use bare URLs. Here is the [rule](https://github.com/updownpress/markdown-lint/blob/master/rules/034-no-bare-urls.md). Basically, the URL isn't converted to a link in many parsers without the angular brackets.

Reply:
Fair enough, Markdownlint has some nice rules, but we use the [GFM (GitHub-flavoured Markdown)](https://github.github.com/gfm/) and [MyST Markdown](https://mystmd.org/) variants for rendering on GitHub and our Sphinx docs, which tie in nicely with where these files are hosted.

These changes are welcome indeed and the cleanup is well-deserved, but I would prefer if you could open a separate PR that adds Markdownlint as a pre-commit hook, so that this PR is limited to just build/packaging infra changes.